### PR TITLE
Proxy logic for bids

### PR DESCRIPTION
**Pull request by PhABC**
When Syndicate can participate in an auction, a Bidder contract will be created and bid will be submitted. 

If we get outbid or we win, we can withdraw funds back to Syndicate.

### CODE REVIEW

File: contracts/Bidder.sol [hunk 1]
```diff
@@ -16,32 +16,51 @@ contract Bidder is SimpleFlopper {
...
+    require(bidSubmited, "Bidder/finalize: Bid not yet submitted");
+
+    // If auction was finalized, end should be 0x0.
+    ( , ,address bidder, , uint48 end) = SimpleFlopper.getCurrentBid(auctionID);
+
+    // If auction isn't closed, we try to close it ourselves
+    if (end != 0) {
+      // If we are the winning bidder, we finalize the auction
+      // Otherwise we got outbid and we withdraw DAI
+      if (bidder == address(this)) {
```
Comment (0age) [inline contracts/Bidder.sol]: Are we sure we don't need to finalize the auction in the event we lost?

File: contracts/DaiBackstopSyndicate.sol [hunk 2]
```diff
@@ -107,69 +106,56 @@ contract DaiBackstopSyndicate is DaiBackstopSyndicateInterface, SimpleFlopper, E
...
 
     // Ensure that the auction in question has not already been entered
     require(
-      !_activeAuctions.contains(auctionId), "Already participating in this auction"
+      _bidders[auctionId] == address(0x0), 
+      "DaiBackstopSyndicate/enterAuction: Already participating in this auction"
     );
 
     // Determine the Dai currently held by the contract.
+    // ??? Probably not necessary since this should throw in the _bid() call
```
Comment (0age) [inline contracts/DaiBackstopSyndicate.sol]: Yeah, plus we want to use `sump` anyway right?

File: contracts/DaiBackstopSyndicate.sol [hunk 3]
```diff
@@ -107,69 +106,56 @@ contract DaiBackstopSyndicate is DaiBackstopSyndicateInterface, SimpleFlopper, E
...
+      daiBalance >= 50000 * 1e18, 
+      "DaiBackstopSyndicate/enterAuction: Insufficient Dai available for auction."
     );
 
-    // Ensure that the current bid is not at a higher price than backstop.
-    (uint256 currentDaiBid, uint256 curentLotSize, , , ) = SimpleFlopper.getCurrentBid(auctionId);
+    // Create auction's Bidder contract and approve it for VAT 
+    Bidder bidder = new Bidder(SimpleFlopper.getFlopperAddress(), auctionId);
+    _bidders[auctionId] = address(bidder);
+    _VAT.hope(address(bidder));
```
Comment (0age) [inline contracts/DaiBackstopSyndicate.sol]: @samczsun what's the plain-english explanation of what this one does?

File: contracts/Bidder.sol (hunk 1 above)
Comment (PhABC) [reply contracts/Bidder.sol]: If someone outbid us, I believe they send back DAI to our Bidder right away as per https://github.com/makerdao/dss/blob/44330065999621834b08de1edf3b962f6bbd74c6/src/flop.sol#L136

File: contracts/DaiBackstopSyndicate.sol (hunk 3 above)
Comment (PhABC) [reply contracts/DaiBackstopSyndicate.sol]: > We create our Auction Vault and enable it to pull DAI from the syndicate in order to bid

Here's my second attempt :upside_down_face:

**Pull request merged**